packages/aws-rfdk/lib/deadline/lib/configure-spot-event-plugin.ts

```diff
@@ -0,0 +1,491 @@
+            ],
+            resources: roles.length !== 0 ? roles : undefined,
+            conditions: {
+              StringLike: {
+                'iam:PassedToService': 'ec2.amazonaws.com',
+              },
+            },
+          }),
+          new PolicyStatement({
+            actions: [
+              'ec2:CreateTags',
+            ],
+            resources: ['arn:aws:ec2:*:*:spot-fleet-request/*'],
+          }),
+        ],
+        roles: [
+          props.renderQueue.grantPrincipal as Role,
+        ],
+      });
+    }
+
+    const region = Construct.isConstruct(props.renderQueue) ? Stack.of(props.renderQueue).region : Stack.of(this).region;
+    const openSslLayerArns: any = ARNS['openssl-al2'];
+    const openSslLayer = LayerVersion.fromLayerVersionArn(this, 'OpenSslLayer', openSslLayerArns[region]);
+
+    const configurator = new LambdaFunction(this, 'Configurator', {
+      vpc: props.vpc,
+      vpcSubnets: props.vpcSubnets ?? { subnetType: SubnetType.PRIVATE },
+      description: `Used by a ConfigureSpotEventPlugin ${this.node.addr} to perform configuration of Deadline Spot Event Plugin`,
+      code: Code.fromAsset(path.join(__dirname, '..', '..', 'lambdas', 'nodejs'), {
+      }),
+      environment: {
+        DEBUG: 'false',
+      },
+      runtime: Runtime.NODEJS_12_X,
+      handler: 'configure-spot-event-plugin.configureSEP',
+      layers: [ openSslLayer ],
+      timeout: Duration.minutes(2),
+      logRetention: RetentionDays.ONE_WEEK,
+    });
+
+    configurator.connections.allowToDefaultPort(props.renderQueue);
+    props.caCert?.grantRead(configurator.grantPrincipal);
+
+    const pluginConfig: InternalSpotEventPluginSettings = {
+      awsInstanceStatus: props.configuration?.awsInstanceStatus ?? SpotEventPluginAwsInstanceStatus.DISABLED,
+      deleteEC2SpotInterruptedWorkers: props.configuration?.deleteEC2SpotInterruptedWorkers ?? false,
+      deleteSEPTerminatedWorkers: props.configuration?.deleteSEPTerminatedWorkers ?? false,
+      idleShutdown: props.configuration?.idleShutdown?.toMinutes({integral: true}) ?? 10,
+      loggingLevel: props.configuration?.loggingLevel ?? SpotEventPluginLoggingLevel.STANDARD,
+      preJobTaskMode: props.configuration?.preJobTaskMode ?? SpotEventPluginPreJobTaskMode.CONSERVATIVE,
+      region: props.configuration?.region ?? region,
+      enableResourceTracker: props.configuration?.enableResourceTracker ?? true,
+      maximumInstancesStartedPerCycle: props.configuration?.maximumInstancesStartedPerCycle ?? 50,
+      state: props.configuration?.state ?? SpotEventPluginState.DISABLED,
+      strictHardCap: props.configuration?.strictHardCap ?? false,
+    };
+    const spotFleetRequestConfigs = this.mergeSpotFleetRequestConfigs(props.spotFleets);
+
+    const properties: SEPConfiguratorResourceProps = {
+      connection: {
+        hostname: props.renderQueue.endpoint.hostname,
+        port: props.renderQueue.endpoint.portNumber.toString(),
```

Comment:
I believe we should be using `props.renderQueue.endpoint.portAsString()` ([ref docs](https://docs.aws.amazon.com/rfdk/api/latest/docs/aws-rfdk.Endpoint.html#port-wbr-as-wbr-string)). This is because if the port is an unresolved token (e.g. reference to a resource's output attribute), CDK will not replace it when embedded in a string.

Reply:
Changed, thanks!